src/frontc/cabs2cil.ml

```diff
@@ -4269,76 +4263,75 @@ and doExp (asconst: bool)   (* This expression is used as a constant *)
               fv.vname = "__builtin_choose_expr" || fv.vname = "__builtin_tgmath"
             | _ -> false
         in
-
+        let isBuiltinNan =
+          match f'' with
+          | Lval (Var fv, NoOffset) -> fv.vname = "__builtin_nan" || fv.vname = "__builtin_nanf" ||
+            fv.vname = "__builtin_nanl" || fv.vname = "__builtin_nans" || fv.vname = "__builtin_nansf" || fv.vname = "__builtin_nansl"
+          | _ -> false
+        in
+        if isBuiltinNan && asconst then
+          (* Replace call to builtin nan with computation yielding NaN *)
+          let onef = Const(CReal(0.0,FDouble,None)) in
+          let zerodivzero = mkCast (BinOp(Div,onef,onef,doubleType)) resType in
+          (empty,zerodivzero,resType)
+        else (
```

Comment:
This is the only semantic diff, the rest is just due to the crazy indenting CIL had that I needed to fix to wrap my head around things.